<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)"/>

  <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch"/>

  <!-- nodelet settings-->
  <arg name="standalone" default="false" />
  <arg name="manager" default="$(arg UAV_NAME)_vio_imu_manager" />

  <!-- <arg name="manager" default="$(arg UAV_NAME)_vinsimufilter_manager" /> -->
  <arg name="n_threads" default="8" />
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

    <!--//{ NODELET MANAGER -->
  <group ns="$(arg UAV_NAME)">
    <node if="$(eval arg('standalone') == 0)" pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen" launch-prefix="$(arg launch_prefix)">
      <param name="num_worker_threads" value="8" />
    </node>
  </group>
  <!--//}-->

  <group ns="$(arg UAV_NAME)">

    <node pkg="nodelet" type="nodelet" name="vins_imu_filter" args="$(arg nodelet) mrs_vins_imu_filter/VinsImuFilter $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix)" output="screen">

      <rosparam file="$(find mrs_vins_imu_filter)/config/icm_42688.yaml" command="load" />

      <param name="uav_name" value="$(arg UAV_NAME)"/>

      <remap from="~imu_in" to="vio_imu/imu_raw"/>
      <remap from="~imu_out" to="vio_imu/imu_filtered"/>
      <remap from="~accel_in" to="none"/>
      <remap from="~gyro_in" to="none"/>

    </node>

  </group>

</launch>
